core/state_processor.go

```diff
@@ -263,6 +292,40 @@ func ProcessParentBlockHash(prevHash common.Hash, evm *vm.EVM) {
 	evm.StateDB.Finalise(true)
 }
 
+// ProcessL1OriginBlockInfo stores the L1 block info in the L1OriginSource contract
+// as defined in RIP-7859.
+func ProcessL1OriginBlockInfo(l1OriginSource *L1OriginSource, evm *vm.EVM) {
+	if tracer := evm.Config.Tracer; tracer != nil {
+		onSystemCallStart(tracer, evm.GetVMContext())
+		if tracer.OnSystemCallEnd != nil {
+			defer tracer.OnSystemCallEnd()
+		}
+	}
+
+	msg := &Message{
+		From:      params.SystemAddress,
+		GasLimit:  30_000_000,
+		GasPrice:  common.Big0,
+		GasFeeCap: common.Big0,
+		GasTipCap: common.Big0,
+		To:        &params.L1OriginContractAddress,
+		Data:      l1OriginSource.UpdateL1OriginSourceCallData(),
+	}
+
+	evm.SetTxContext(NewEVMTxContext(msg))
+	evm.StateDB.AddAddressToAccessList(params.L1OriginContractAddress)
+	_, _, err := evm.Call(msg.From, *msg.To, msg.Data, msg.GasLimit, common.U2560)
+
+	if err != nil {
+		panic(fmt.Errorf("failed to process L1 block info: %v", err))
+	}
+
+	if evm.StateDB.AccessEvents() != nil {
+		evm.StateDB.AccessEvents().Merge(evm.AccessEvents)
+	}
+	evm.StateDB.Finalise(true)
+}
```

Comment:
I'd move this and all related code `UpdateL1OriginSourceCallData/L1OriginSource` to separate file so it's easier to maintain (especially with keeping up to date with the upstream and merging with other `L2-geth` implementations.)

I suggest `state_processor_rollup.go`.
I'd do the same for the test, i.e. `verkle_witness_rollup_test.go`